src/many-ledger/src/module/ledger_tokens.rs

```diff
@@ -0,0 +1,197 @@
+use crate::error;
+use crate::migration::tokens::TOKEN_MIGRATION;
+use crate::module::account::verify_account_role;
+use crate::module::LedgerModuleImpl;
+use crate::storage::LedgerStorage;
+use many_error::ManyError;
+use many_identity::Address;
+use many_modules::account;
+use many_modules::account::features::TryCreateFeature;
+use many_modules::account::Role;
+use many_modules::ledger::{
+    LedgerTokensModuleBackend, TokenAddExtendedInfoArgs, TokenAddExtendedInfoReturns,
+    TokenCreateArgs, TokenCreateReturns, TokenInfoArgs, TokenInfoReturns,
+    TokenRemoveExtendedInfoArgs, TokenRemoveExtendedInfoReturns, TokenUpdateArgs,
+    TokenUpdateReturns,
+};
+use many_types::Either;
+
+fn verify_tokens_acl(
+    storage: &LedgerStorage,
+    sender: &Address,
+    addr: &Address,
+    roles: impl IntoIterator<Item = Role>,
+) -> Result<(), ManyError> {
+    if addr != sender {
+        if let Some(account) = storage.get_account(addr) {
+            verify_account_role(
+                &account,
+                sender,
+                account::features::tokens::TokenAccountLedger::ID,
+                roles,
+            )?;
+        } else {
+            return Err(error::unauthorized());
+        }
+    }
+    Ok(())
+}
+
+impl LedgerTokensModuleBackend for LedgerModuleImpl {
+    fn create(
+        &mut self,
+        sender: &Address,
+        args: TokenCreateArgs,
+    ) -> Result<TokenCreateReturns, ManyError> {
+        if !self.storage.migrations().is_active(&TOKEN_MIGRATION) {
+            return Err(ManyError::unknown(
```

Comment:
You should return `InvalidMethodName` instead, which is more expected if a method isn't implemented.